[dominoswiss] Initial contribution (#9837)
openhab/openhab-addons: bundles/org.openhab.binding.dominoswiss/README.md

# Dominoswiss Binding

This binding allows the control of rollershutters, using an eGate as gateway and Dominoswiss radio receivers. 
The eGate-gateway is connected via ethernet to openHAB and sends its commands via radio to all rollershutters. 
See https://www.brelag.com/ for more information.

## Supported Things

eGate: Dominoswiss eGate Server. The eGate is the gateway which sends the commands to the connected rollershutters. The bridge-type ID is egate.
Blind: represents one rollershutter, that can be controlled via eGate. The thing-type ID is blind.

## Discovery

Unfortunately no automatic discovery is possible due to protocol restrictions. 
Every rollershutter must be known by eGate and can be called by it's number of storage-place on eGate gateway.

## Thing Configuration

The bridge "eGate" has one channel "getconfig" which returns the config of this bridge. 
The eGate is configured with both an `ipAddress` and a port.
The `ipAddress` might be in format like 192.168.1.1 or as the hostname representing the eGate in your network.

> Port is represented as a number.

Reviewer: As you added a table, these could be removed.

# Dominoswiss Binding

> This binding allows the controll of rollershutters, using an egate as gateway and dominoswiss radio receivers. The egate-gateway is connectet via ethernet to openhab and sends it's commands via radio to all rollershutters. See https://www.brelag.com/ for more information.

Reviewer 2: ```suggestion
This binding allows the control of rollershutters, using an egate as gateway and Dominoswiss radio receivers. 
The egate-gateway is connected via ethernet to openHAB and sends its commands via radio to all rollershutters.
See https://www.brelag.com/ for more information.
```

## Supported Things

> EGate: Dominoswiss EGate Server. The egate is the gateway which sends the commands to the connected rollershutters. 

Reviewer 2: Two things here...

Please put each sentence on its own line.
Which is the correct way? EGate vs egate, then use the correct one right through the readme.

Blind: represents one rollershutter, that can be controlled via EGate

## Discovery

> Unfortunately no automatic discovery is possible due to protocol restrictions. Every rollershutter must be known by egate and can be called by it's number of storage-place on egate gateway.

Reviewer 2: ```suggestion
Unfortunately no automatic discovery is possible due to protocol restrictions.
Every rollershutter must be known by egate and can be called by it's number of storage-place on egate gateway.
```
egate or EGate? Moved each sentence onto its own line as above suggestion.

## Binding Configuration

> _If your binding requires or supports general configuration settings, please create a folder ```cfg``` and place the configuration file ```<bindingId>.cfg``` inside it. In this section, you should link to this file and provide some information about the options. The file could e.g. look like:_

Reviewer 2: Does this apply to your binding? If it does, why not use an example for your binding and get rid of Hue example?

```
# Configuration for the Philips Hue Binding
#
# Default secret key for the pairing of the Philips Hue Bridge.
# It has to be between 10-40 (alphanumeric) characters
# This may be changed by the user for security reasons.
secret=openHABSecret
```

_Note that it is planned to generate some part of this based on the information that is available within ```src/main/resources/OH-INF/binding``` of your binding._

> _If your binding does not offer any generic configurations, you can remove this section completely._

Reviewer 2: I recommend you remove it completely as this looks like it was auto generated a very long time ago by an old skeleton script ?!?!?

## Thing Configuration

The bridge "egate" has one channel "getconfig" which returns the config of this bridge. 

> The egate is configured with both an ipaddress and a port. The ip address might be in format like 192.168.1.1 or as the hostname representing the egate in your network.  Port is represented as a number.`

Reviewer 2: ```suggestion
The egate is configured with both an `ipAddress` and a port.
The `ipAddress` might be in format like 192.168.1.1 or as the hostname representing the egate in your network.
Port is represented as a number.`
```

```java
Bridge dominoswiss:egate:myEgate "My EGate Server" @ "Home" [ ipAddres="localhost", port="5700" ]
```

The thing blind represents one blind on the egate. Each blind is represented by an id set on your egate.
  
```java 
Thing blind officeBlind "Office" @ "1stFloor" [ id="1"]
```

The blind-Thing has the following channels:
| Channel Type ID  | Item Type | Description                                                                                       |   |   |
|------------------|-----------|---------------------------------------------------------------------------------------------------|---|---|

> | pulseUp            |Blind      |sends one pulse up to this blind.|

Reviewer 2: Is the item type 'Rollershutter' instead of Blind?

| pulsedown          |Blind      |sends one pulse down to this blind|
| continuousup       |Blind      |sends a continuous up to this blind. The blind will automatically stop as it is fully up.|
|continuousdown      |Blind      | send a continous down to this blind. The blind will automatically stop as it is fully down.|
| stop               |Blind      | stop the action of the blind. The command will be imadiatly sent to the blind.|
| shutter            |Blind      |this is used to bind the channel to the shutter item. There are no further rules needed this channel will handel the up/down/stop commands. See example for usage.|
| tilt               |Blind      | same as shutter, this will handel all up/down/stop - tilt commands to be used with the shutter-item.| 
| tiltup             |Blind      | sends 3 pulse-up commands to this blind to tilt the blind. If your blind needs more than 3 pulse-up, create a rule yourself with three pluse-up commands. Between each pulse-up you should wait 150ms to let the command be processed. 

> | tiltdown           |Blind      |sends 3 pulse-down commands to this blind to tilt the blind. If your blind needs more than 3 pulse-down, create a rule yourself with three pluse-down commands. Between each pulse-down you should wait 150ms to let the command be processed. |

Reviewer 2: The channels should use the Java naming called lowerCamelCase. See....

https://www.openhab.org/docs/developer/guidelines.html#java-coding-style

Author: Thank you @Skinah for your input, I fixed that in my new commit
